### PR TITLE
Fix #14, use json to store icons

### LINKED ISSUE #14: [advice] Use json to store `g:nerdfont#path#{extension|pattern|basename}#defaults`
1. It is easier to edit even for those contributors is not familiar with vim script
2. It is easier to compare the difference between this project's icon and other projects which provide icon display, so we can add icons more easily.
3. It is easier to sort json by `jq`

Just my 2c, TIA!

### PR DESCRIPTION
**Pull request by Freed-Wu**
According to https://github.com/ogham/exa/blob/master/src/output/icons.rs
add some icons

sort all json alphabetically

Because https://github.com/neovim/neovim/issues/20757
Use '.' to replace '' to get default value

And, add icons for some directory, like exa:

Screenshot from 2022-10-21 14-01-50

### CODE REVIEW

**Comment by Freed-Wu** [comment]
> Error: Command './configure --prefix=/Users/runner/vim-v8.2.0235 --with-features=huge --enable-fail-if-missing' exited non-zero status 1: configure: error: WRONG!  uint32_t not defined correctly. 

Some test failed. Why?

**Review by lambdalisue — CHANGES_REQUESTED** [review]
~~Please split and organize your commit logically so that I can review the changes.~~

I'm sorry I thought multiple things are in one commit because PR describe several things but it seems things are quite small and can be put into a single commit. I'll review the changes.

**Comment by lambdalisue** [comment]
> Some test failed. Why?

Never mind. It seems the failure is not caused by this PR but caused by GitHub macOS runner chage. I'll fix it

File: autoload/nerdfont/path/pattern.vim [hunk 1]
```diff
@@ -1,21 +1,11 @@
...
-      \ '.*require.*\.js$'      : '',
-      \ '.*vimrc.*'             : '',
-      \ 'Vagrantfile$'          : '',
-      \}
+let g:nerdfont#path#pattern#defaults = nerdfont#get_json('pattern')
 
 function! nerdfont#path#pattern#find(path) abort
   for [k, v] in s:m
-    if a:path =~# k
+    if a:path =~# '\v' . k
```
Comment (lambdalisue) [inline autoload/nerdfont/path/pattern.vim]: Please do not use "very magic". I prefer to force developers to use Vim's regex.

**Review by lambdalisue — CHANGES_REQUESTED** [review]

File: autoload/nerdfont/path/pattern.vim (hunk 1 above)
Comment (Freed-Wu) [reply autoload/nerdfont/path/pattern.vim]: OK, I'll fix it. I add `\v` just due to I am afraid someone change `&magic` in their vimrc. Should we change `\v` to `\m`?

Comment (lambdalisue) [reply autoload/nerdfont/path/pattern.vim]: That's very good idea.

**Review by lambdalisue — APPROVED** [review]
LGTM

**Comment by lambdalisue** [comment]
Released. https://github.com/lambdalisue/nerdfont.vim/releases/tag/v1.4.0 Thanks for your contribution 🎉

**Comment by Freed-Wu** [comment]
You are welcome.

I still have some advice: Now the json is sorted alphabetically. If you think
sorted is better than unsorted, how about add a test to check if it is sorted
after every PR? If we don't have any test to ensure the sort, after many PRs,
they may be changed to unsorted.

**Comment by lambdalisue** [comment]
> I still have some advice: Now the json is sorted alphabetically. If you think
sorted is better than unsorted, how about add a test to check if it is sorted
after every PR? If we don't have any test to ensure the sort, after many PRs,
they may be changed to unsorted.

I also worry about that. Could you add such CI and send me a PR? (I'll do otherwise but it would take time)

**Comment by Freed-Wu** [comment]
> Could you add such CI

I'll take a try.

**Pull request merged**